### PR TITLE
Support Twitter oauth login

### PR DESCRIPTION
**Pull request by chyt**
According to [Firebase docs](https://cloud.google.com/identity-platform/docs/reference/rest/v1/accounts/signInWithIdp), OAuth token secret is required to support logging in with Twitter using an OAuth 1.0 credential:

> If the user is signing in to the Twitter provider using a Twitter OAuth 1.0 credential, this should be set to accessToken=[TWITTER_ACCESS_TOKEN]&oauthTokenSecret=[TWITTER_TOKEN_SECRET]&providerId=twitter.com, where [TWITTER_ACCESS_TOKEN] and [TWITTER_TOKEN_SECRET] should be replaced with the Twitter OAuth access token and Twitter OAuth token secret respectively.

This PR adds the ability to specify such a secret when signing in using an IDP access token. Unfortunately there are no tests, but hopefully this will provide the basic implementation for this capability.

### CODE REVIEW

File: src/Firebase/Auth/SignInWithIdpCredentials.php [hunk 1]
```diff
@@ -31,6 +34,16 @@ public static function withAccessToken(string $provider, string $accessToken): s
         return $instance;
     }
 
+    public static function withAccessTokenAndOauthTokenSecret(string $provider, string $accessToken, string $oauthTokenSecret): self
+    {
+        $instance = new self();
+        $instance->provider = $provider;
+        $instance->accessToken = $accessToken;
```
Comment (jeromegamez) [inline src/Firebase/Auth/SignInWithIdpCredentials.php:41]:
```suggestion
        $instance = self::withAccessToken($provider, $accessToken);
```

**Review by jeromegamez — CHANGES_REQUESTED** [review]

File: src/Firebase/Auth.php [hunk 2]
```diff
@@ -779,7 +778,7 @@ public function signInAnonymously(): SignInResult
      *
      * @throws FailedToSignIn
      */
-    public function signInWithIdpAccessToken($provider, string $accessToken, $redirectUrl = null): SignInResult
+    public function signInWithIdpAccessToken($provider, string $accessToken, string $oauthTokenSecret = null, $redirectUrl = null): SignInResult
```
[the remark this answers is not in the document]
Comment (chyt) [reply src/Firebase/Auth.php]: Fixed

File: src/Firebase/Auth/SignInWithIdpCredentials.php (hunk 1 above)
Comment (chyt) [reply src/Firebase/Auth/SignInWithIdpCredentials.php]: Fixed

File: src/Firebase/Auth.php [hunk 3]
```diff
@@ -212,8 +212,7 @@ public function createUserWithEmailAndPassword($email, $password): UserRecord
     {
         return $this->createUser(Request\CreateUser::new()
             ->withUnverifiedEmail($email)
-            ->withClearTextPassword($password)
-        );
+            ->withClearTextPassword($password));
```
[the remark this answers is not in the document]
Comment (chyt) [reply src/Firebase/Auth.php]: Running that command changes a bunch of additional files. I ended up manually reverting the linespace changes.

**Comment by jeromegamez** [comment]
Good work!  🙌  I merged your changes with 37f5ae9d78b00d78d2eb32b4a4f641d118a6bf86 (I just made the last CS change and added a test, while making sure that you're associated with the commit) and [added some more convenience methods](https://github.com/kreait/firebase-php/commit/d7d40edf6068069f4464607fa2bbf7b7b5624c88) while I was at it 😅 

Thank you! 🌺

**Pull request closed without merging**